Initial implementation of SurplusAuction component and stories

aomafarag opened the pull request:
Contributes to #327 

![image_2022-07-04_13-44-10](https://user-images.githubusercontent.com/85403429/177148198-87349695-1443-4ebe-b843-6193f1bad8d1.png)

Checklist:
- [x] issue number linked above after pound (`#`)
  - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [x] issue checkboxes are all addressed
- [x] manually checked my feature / not applicable
- [ ] wrote tests / not applicable
- [x] attached screenshots / not applicable

aomafarag commented:
Types, helper functions and some components which are necessary for a correct and fully functional implementation are still not merged into main. Examples: helper function to generate a fake **surplus** auction, proper `SurplusAuction` type, etc.

LukSteib commented:
Can you please provide proper description in all of your PRs for the top level commands acoording to our standards. 
Take the following PR as an example and/or call @fegbert for a quick intro

aomafarag commented:
> Can't you piggyback on the fake data functions used for other components and do it here respectively?

OK, I did that in my most recent comment.

> Are the currently defined types for `SurplusAuction` not sufficient? If so, please specify what exactly you are missing here

Yes, they're not sufficient. The original `Auction` component implements the `AuctionTranscation` interface, which has several properties (inherited from parent interfaces) , e.g. `isFinished`, which are necessary, e.g. to check that the auction is over. The `SurplusAuction` interface doesn't implement the same parent interfaces, nor does it have their properties. Moreover, the logic to calculate the highest bid is missing because `SurplusAuction` here doesn't implement `SurplusEvent`, and therefore the `latestBid` cannot be retrieved.

> When it comes to other components like `SurplusAuctionRestartPanel` you might neglect this one for now or base this PR on the one you need.

I neglected it for now.

aomafarag commented:
I rebased this branch on [the surplus auction restart panel one](https://github.com/sidestream-tech/unified-auctions-ui/tree/sa-restart-panel). I also updated the logic to match the surplus auction interface. However, `areAuctionsFetching`, `fetchedAt`, etc. need to be updated with a functions/properties relevant to surplus auctions. I also added two new optional properties, `marketUnitPrice` and `marketUnitPriceToUnitPriceRatio`, to `SurplusAuctionActive`.

LukSteib commented:
Not able to start storybook (see console error below)

```
ERROR in ./components/SurplusAuction.vue?vue&type=script&lang=ts& (./node_modules/vue-docgen-loader/lib??ref--12!./node_modules/ts-loader??ref--4-0!./node_modules/vue-loader/lib??vue-loader-options!./components/SurplusAuction.vue?vue&type=script&lang=ts&)
Module not found: Error: Can't resolve '~/components/panels/SurplusAuctionRestartPanel.vue' in '/Users/lukas-admin/unified-auctions-ui/frontend/components'
 @ ./components/SurplusAuction.vue?vue&type=script&lang=ts& (./node_modules/vue-docgen-loader/lib??ref--12!./node_modules/ts-loader??ref--4-0!./node_modules/vue-loader/lib??vue-loader-options!./components/SurplusAuction.vue?vue&type=script&lang=ts&) 9:0-92 14:8-34
 @ ./components/SurplusAuction.vue?vue&type=script&lang=ts&
 @ ./components/SurplusAuction.vue
 @ ./components/SurplusAuction.stories.js
 @ ./components sync ^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.(js|jsx|ts|tsx))$
 @ ./.storybook/generated-stories-entry.js
 @ multi ./node_modules/@storybook/core-client/dist/esm/globals/polyfills.js ./node_modules/@storybook/core-client/dist/esm/globals/globals.js ./.storybook/storybook-init-framework-entry.js ./node_modules/@storybook/addon-docs/dist/esm/frameworks/common/config.js-generated-other-entry.js ./node_modules/@storybook/addon-docs/dist/esm/frameworks/vue/config.js-generated-other-entry.js ./node_modules/@storybook/addon-links/dist/esm/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addArgs.js-generated-other-entry.js ./node_modules/@storybook/addon-backgrounds/dist/esm/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-backgrounds/dist/esm/preset/addParameter.js-generated-other-entry.js ./node_modules/@storybook/addon-measure/dist/esm/preset/preview.js-generated-other-entry.js ./node_modules/storybook-addon-outline/dist/esm/preset/addDecorator.js-generated-other-entry.js ./node_modules/storybook-tailwind-dark-mode/dist/preset/preview.js-generated-other-entry.js ./.storybook/preview.js-generated-config-entry.js ./.storybook/generated-stories-entry.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined
```

aomafarag commented:
Storybook is up and running now. I also reused `AuctionRestartPanel` instead of `SurplusAuctionRestartPanel`, which is yet to be merged, and upon which this PR has been based. However, certain values such as `Auction Price`, `Market Difference`, `Last updated`, etc. cannot be displayed unless they are fetched properly. I suppose @KirillDogadin-std is working on that?

fegbert commented on core/src/surplus.ts:
```diff
@@ -0,0 +1,88 @@
+import type { SurplusAuction, SurplusAuctionBase } from './types';
```
Can you try updating this branch from `main`? It seems to be outdated and therefore adding whole files that already exist

fegbert commented:
![image](https://user-images.githubusercontent.com/47861849/177781212-a4a342b5-a31f-4556-a741-0a1987aeb33a.png)
...well thats not good

I think `sa-restart-panel` is also outdated, so please update that one as well 😅

aomafarag commented:
> (and then check if the file amount got less) 😅

@fegbert it's still the same 😅

zoey-kaiser commented:
This PR still uses an outdated version of the `generateFakeSurplusAuction` helper. Please pull main into `sa-restart-panel` and then update all branches going into it to ensure everything is correct 🤗

zoey-kaiser commented:
In call we agreed to close this PR, as it contains many old files. @aomafarag will open a new PR with the changes needed, to ensure everything is a bit cleaner 😄.
